jaxrs-api/src/main/java/jakarta/ws/rs/ext/FactoryFinder.java

```diff
@@ -168,4 +163,44 @@ private static Object newInstance(final String className, final ClassLoader clas
         throw new ClassNotFoundException(
                 "Provider for " + factoryId + " cannot be found", null);
     }
+
+    private static ClassLoader getClassLoader() {
+        if (System.getSecurityManager() == null) {
+            return FactoryFinder.class.getClassLoader();
+        }
+        return AccessController.doPrivileged((PrivilegedAction<ClassLoader>) FactoryFinder.class::getClassLoader);
+    }
+
+    private static <T> T findFirstService(final String factoryId, final ClassLoader cl, final Class<T> service) {
+        final FindServiceAction<T> action = new FactoryFinder.FindServiceAction<>(factoryId, cl, service);
+        if (System.getSecurityManager() == null) {
+            return action.run();
+        }
+        return AccessController.doPrivileged(action);
```

Comment:
Same comment as above. Can we avoid the doPriv on the `ServiceLoader.load(...)` calls?